File: DSAinC/ArrayInsertion.c
#include <stdio.h>

int main(){

    // An algorithm to insert value 18 at index 2 in the numbers array

    int numbers[7] = {23,2,56,34,13};  //The unfilled array
    int L = 0; // The lower index
    int U = 6; // The upper index
    int n = 5; //The number of elements in the array
    int size = 7; //Size of the array

    int i; // The target
    int location = 2; // insertion index
    int value = 18; // insertion value

     printf("Array before insertion \t");
    for(int a = 0; a < n; a++){
        printf("%d,",numbers[a]);
    }

    printf("\n");

    if(size <= n ){
        printf("The array is full. Insertion not possible");  // Check if array is filled
    }
    else{
        i = n - 1;  //start at the index of the last value
        while(i >= location){
            numbers[i+1] = numbers[i];  // Push elements until the location
            i = i-1;
        }
        numbers[location] = value;      //Insert vale
        n = n + 1;                      // Increase the upper index
    }

    printf("Array after insertion \t"); 
    for(int a = 0; a < n; a++){ 
        printf("%d,",numbers[a]);
    }

    return 0;
}

File: DSAinJava/ArrayInsertion.java
public class ArrayInsertion {
    // Attempt to insert a value into an array
    public static void main(String[] args) {
        int[] numbers = new int[7]; // The array of size 7
        numbers[0] = 23;
        numbers[1] = 2;
        numbers[2] = 56;
        numbers[3] = 34;
        numbers[4] = 13;

        int size = 7; //Size of the array
        int n = 5; // number of elements in the array
        int i; //The target

        int location = 2; //Insertion location
        int value = 18; //Insertion value

        System.out.println("Array before insertion");
        for (int a = 0; a < n; a++) {
            System.out.print(numbers[a] + ",");
        }
        if(size <= n){
            System.out.println("The array is full. No insertion allowed");
        }
        else {
            i = n - 1; //Start from the last element index
            while (i >= location) {
                numbers[i + 1] = numbers[i]; //Push values until the location
                i--;
            }
            numbers[location] = value; //Insert the value at the location
            n = n + 1;
        }

        System.out.println("\nArray after insertion");
        for (int a = 0; a < n; a++) {
            System.out.print(numbers[a] + ",");
        }
    }
}
